.github/workflows/upload_component.yml

```
@@ -65,4 +72,7 @@ jobs:
             zlib;
             libpng;
           namespace: "espressif"
+          # API token will only be available in the master branch.
+          # However, dry-run doesn't require a valid token.
           api_token: ${{ secrets.IDF_COMPONENT_API_TOKEN }}
+          dry_run: ${{ github.ref_name != 'master' }}
```

This should also run with --dry-run argument in forks (similar change on line 21)
```suggestion
          dry_run: ${{ github.ref_name != 'master'  && github.repository_owner == 'espressif' }}
```

---

Thanks, applied!